src/plugins/Toast/Toast.ts

```diff
@@ -1,74 +1,107 @@
+   
+  if (!element) {
+      throw new Error(
+        "Toast plugin wasn't provided a valid mount point. Make sure the mount point exists in the DOM or pass nothing to create one automatically."
+      )
+  }
+
+  return element
+}
+
+function getMountElement(mountPoint: Element | string | undefined): Element {
+  if(!mountPoint){
+    return createDefaultMountElement()
+  }
+
+  if(typeof mountPoint === 'string'){
+    return tryQuerySelector(mountPoint)
+  }
+
+  return mountPoint
+}
+
+let toastId = 0
+const queue: Toast[] = reactive([])
+
+function getToastId(): number {
+  return toastId++
 }
 
-export interface ToastPlugin {
-  install(app: App): void
+function hideToast(id: number): void{
+  const index = queue.findIndex(toast => toast.id === id)
+
+  if (index > -1) {
+    queue.splice(index, 1)
+  }
 }
 
-export const injectionKey: InjectionKey<ToastPlugin> = Symbol('toast')
-
-export default {
-  install: (app: App, options: any = {}) => {
-    let mountPoint: Element
-
-    if (options.mountPoint) {
-      mountPoint = document.querySelector(options.mountPoint)
-      if (!mountPoint)
-        throw new Error(
-          "Toast plugin wasn't provided a valid mount point. Make sure the mount point exists in the DOM or pass nothing to create one automatically."
-        )
-    } else {
-      mountPoint = document.createElement('div')
-      mountPoint.id = 'miter-toast-app'
-      document.body.appendChild(mountPoint)
-    }
-
-    const toastApp = createApp(ToastContainer)
-
-    const toastContainer: any = toastApp.mount(mountPoint)
-
-    const Toast = {
-      /**
-       * Adds a new toast to the global interface - if a component is passed, the content string will be ignored
-       * @param options: { component: Component, content: string, color: string, timeout: number }
-       * @returns remove
-       */
-      add(options: ToastOptions = {}): MountedElement {
-        return { remove: toastContainer.add(options) }
-      },
-
-      removeAll(): void {
-        toastContainer.removeAll()
-      }
-    }
-
-    app.config.globalProperties.$toast = Toast
-
-    app.provide('$toast', Toast)
+function showToast(options: ToastOptions): Toast
+function showToast(message: string): Toast
+function showToast(message: string, type: ToastType): Toast
+function showToast(optionsOrMessage: ToastOptions | string, type: ToastType = 'default'): Toast {
+  const id = getToastId()
+
+  const options = typeof optionsOrMessage === 'string'
+    ? { message: optionsOrMessage, type }
+    : optionsOrMessage
+
+  const toast: Toast = { 
+    id, 
+    dismiss: () => hideToast(id), 
+    ...options
   }
+
+  queue.unshift(toast)
+
+  return toast
 }
+
+const ToastPlugin: Plugin = {
+  install: (app: App, options: ToastPluginOptions = {}) => {
+    const element = getMountElement(options.mountPoint)
+    
+    createApp(ToastContainer).mount(element)
+
+    app.config.globalProperties.$toast = showToast
+
+    app.provide('$toast', showToast)
```

Comment:
Assuming this is provide/inject I think this could be removed. If its on the global properties and you can import it directly I don't see a benefit to also being able to inject it.